core/embed/rust/src/ui/model_tr/component/input_methods/wordlist.rs

```diff
@@ -167,7 +171,32 @@ where
             INITIAL_PAGE_COUNTER
         } else {
             let choices_count = <ChoiceFactoryWordlist as ChoiceFactory<T>>::count(&new_choices);
-            get_random_position(choices_count)
+            // There should be always DELETE and at least one letter
+            assert!(choices_count > 1);
+            if choices_count == 2 {
+                // In case there is only DELETE and one letter, starting on that letter
+                // (regardless of the last letter in the textbox)
+                INITIAL_PAGE_COUNTER
+            } else {
+                // We do not want to end up at the same letter as the last one in the textbox
+                loop {
+                    let random_position = get_random_position(choices_count);
+                    let current_action = <ChoiceFactoryWordlist as ChoiceFactory<T>>::get(
+                        &new_choices,
+                        random_position,
+                    )
+                    .1;
+                    if let WordlistAction::Letter(current_letter) = current_action {
+                        if let Some(last_letter) = self.get_last_textbox_letter() {
+                            if current_letter == last_letter {
+                                // Randomly trying again when the last and current letter match
+                                continue;
+                            }
+                        }
+                    }
+                    break random_position;
+                }
+            }
```

Comment:
The function's getting quite long, would it be possible to factor out some of the logic into a function? One possibility would be something like `get_random_position_except(choices_count, current_index)` though i don't know if it's easy to obtain the index here ...